### DISCUSSION TITLE
help to clone a git repo and submodules

### DISCUSSION THREAD

**Discussion by alcarraro** [cpm-cmake/CPM.cmake#614] [Q&A]
How can I use `CPMAddPackage` to replicate the functionality of `git clone --recursive --depth 1 --branch v1.11.2 https://github.com/STMicroelectronics/STM32CubeH7.git`?

I attempted the following:

```cmake
CPMAddPackage(
    NAME STM32CubeH7
    GITHUB_REPOSITORY STMicroelectronics/STM32CubeH7
    GIT_TAG v1.11.2
    DOWNLOAD_ONLY TRUE
)
```

However, this does not download the submodules. Is there a way to include them using `CPMAddPackage`?

**Comment by Avus-c** [comment]
You could try to force a clone with the [GIT_SUBMODULE](https://cmake.org/cmake/help/latest/module/ExternalProject.html#git) argument.
CPM will forward arguments it doesn't know about to FetchContent, which forwards it to ExternalProject.
